v2/apiserver/internal/core/events.go

```diff
@@ -403,13 +419,30 @@ func (e *eventsService) createSingleEvent(
 
 	event.ID = uuid.NewV4().String()
 
+	jobs := []Job{}
 	workerSpec := project.Spec.WorkerTemplate
+	// If the event is a retry of another, defer to the Worker.Spec on the event
+	// itself.  Retain/cache any successful jobs that do not use a shared
+	// workspace.
+	if event.Labels != nil && event.Labels[RetryLabelKey] != "" {
+		workerSpec = event.Worker.Spec
+		for _, job := range event.Worker.Jobs {
+			if job.Status.Phase == JobPhaseSucceeded && !job.UsesWorkspace() {
+				// Capture event ID for logs, if not already set.
```

Comment:
"If not already set"... am I interpretting this correctly that this will do the right thing in the case of a retry of a retry?

Reply:
Indeed.  I'll make this clearer in the comment.

Reply:
Glad you had already thought of that scenario. It hadn't occurred to me until late lastnight. Went to check and you had it covered already!